packages/solid/web/src/index.ts

```diff
@@ -99,9 +99,20 @@ export function Portal<T extends boolean = false, S extends boolean = false>(pro
   return marker;
 }
 
-type DynamicProps<T> = T & {
-  children?: any;
-  component?: Component<T> | string | keyof JSX.IntrinsicElements;
+type ValidElements = keyof JSX.IntrinsicElements;
+type ValidComponent<P> = (props: P) => JSX.Element;
+type ValidConstructor = ValidElements | ValidComponent<unknown> | string;
```

Comment:
I imagine `string` would need to be `string & {}` in order for autocomplete to suggest `ValidElements`. https://tsplay.dev/WY4Mdm (works for `a` but not for `b` where it suggests all tokens).
```suggestion
type ValidConstructor = ValidElements | ValidComponent<unknown> | (string & {});
```

Reply:
The autocomplete works properly (as shown above) so we don't really need that intersection.

Reply:
The autocomplete for `component`, not the rest of the props.

Reply:
Makes sense, I'll do that